python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -607,37 +747,65 @@ async def correction_loop(self):
         the heartbeat frequency."""
 
         while self.summary_state == State.ENABLED:
+
             try:
 
                 # The heartbeat wait is folded here so instead of applying and
                 # then waiting, the loop applies and wait at the same time.
                 corrections_to_apply = []
                 if self.azimuth is not None and self.elevation is not None:
+                    # self.log.debug('Got elevation and azimuth values in correction_loop')
                     elevation = self.elevation
                     azimuth = self.azimuth
 
                     if self.target_elevation is not None and self.target_elevation < self.elevation:
                         # Telescope in going down, need to go ahead and decrease
                         # pressure accordingly
-                        elevation = (self.target_elevation + self.elevation)/2.
+                        elevation = (self.target_elevation + self.elevation) / 2.
                         self.log.debug(f"Telescope going down, getting ahead on correction."
                                        f"el: {self.elevation}, target_el: {self.target_elevation}, "
                                        f"corr_el: {elevation}")
 
                     for correction in self.corrections:
+                        # corrections_routines only has m1, m2 and hexapod, NOT FOCUS
                         if self.corrections[correction] and correction in self.corrections_routines:
                             self.log.debug(f"Adding {correction} correction.")
                             corrections_to_apply.append(
                                 self.corrections_routines[correction](azimuth,
                                                                       elevation))
+
+                    # Check to see if any model offsets need applying due to changes in atspectrograph
+                    # filter/disperser changes, then apply if needed
+                    self.log.debug("self.corrections['atspectrograph']"
+                                   " is {}".format(self.corrections['atspectrograph']))
+                    self.log.debug("self.focus_offset_yet_to_be_applied "
+                                   "is {}".format(self.focus_offset_yet_to_be_applied))
+                    if self.corrections['atspectrograph'] and (self.focus_offset_yet_to_be_applied != 0.0):
```

Comment:
It is not a good idea to check a `float` against `0.0`. You can get rounding errors that will cause this to aways be true.... It is better to add a tolerance and then check that

```
abs(self.focus_offset_yet_to_be_applied) >= tolerance
```

Reply:
Another alternative, if you are controlling the offset the way you are doing is to make `self.focus_offset_yet_to_be_applied = None` once you applied the offset. In this case you would do:

 ```
if self.corrections['atspectrograph'] and self.focus_offset_yet_to_be_applied is not None:
```

Then, once you applied the offset instead of setting `self.focus_offset_yet_to_be_applied = 0.0` (line 788) you would do `self.focus_offset_yet_to_be_applied = None`.

Reply:
I'll do the former, essentially ignoring any offset that is 10x smaller than whatever the hexapod motion tolerance is in `z`.
I think the latter will result in errors later on where it might try to add a value to None...